android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/ReportSiteDomainsRepository.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.cookiebanners
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.engine.gecko.cookiebanners.ReportSiteDomainsRepository.PreferencesKeys.REPORT_SITE_DOMAINS
+import mozilla.components.support.base.log.logger.Logger
+import java.io.IOException
+
+/**
+ * A repository to save reported site domains with the datastore API.
+ */
+class ReportSiteDomainsRepository(
+    private val dataStore: DataStore<Preferences>,
+) {
+
+    companion object {
+        const val SEPARATOR = "@<;>@"
+        const val REPORT_SITE_DOMAINS_REPOSITORY_NAME = "report_site_domains_preferences"
+        const val PREFERENCE_KEY_NAME = "report_site_domains"
+    }
+
+    private object PreferencesKeys {
+        val REPORT_SITE_DOMAINS = stringPreferencesKey(PREFERENCE_KEY_NAME)
+    }
+
+    /**
+     * Check if the given site's domain url is saved locally.
+     * @param siteDomain the [siteDomain] that will be checked.
+     */
+    suspend fun isSiteDomainReported(siteDomain: String) =
```

Comment:
nit: As we are working on library it will be nicer to be explicit with the return type, a now we relay on the from `dataStore.data`, if the signature of changes also `isSiteDomainReported` will change.